Typescript sdk standardization

Superpat opened the pull request:
Fixes a bug in my last pr that was pointed out by @nickchomey and closes #603 

I decided to keep both api's since it was pretty simple.

Maybe wait till @nickchomey has reviewed this since he found out the flaw in the last pr.

Superpat committed:
Bring typescript sdk in line with sdk api's

But still allow passing of stricter types when it makes sense

Superpat committed:
fix bug introduced in https://github.com/starfederation/datastar/pull/698

we want the stream to be ended only if keepalive is not present

nickchomey commented:
Seems ok to me!

bencroker commented:
Thank you both!

While reviewing this I noticed that you’ve hard-coded the Datastar version number in `node.ts` and `deno.ts`. Can you look please into importing the version number instead?

https://github.com/starfederation/datastar/blob/a4c2e874734132b81f1b3d19e222bbf784eb7699/sdk/typescript/src/node/node.ts#L13

https://github.com/starfederation/datastar/blob/a4c2e874734132b81f1b3d19e222bbf784eb7699/sdk/typescript/src/web/deno.ts#L10

Superpat commented:
Right, good point

Superpat committed:
Remove close method from typescript sdk and improve comments

The stream is closed automatically after the callback runs.
The only situation where we want to use keepalive is if we
want an infinite stream, in which case we dont want to close it.

Superpat committed:
improve package.json for typescript sdk

Superpat committed:
Use consts VERSION in typescript sdk tests

Superpat committed:
Improve method comments in typescript sdk

this sdk is also used in javascript where there's less type information

Superpat commented:
@nickchomey I've done some changes to the comments, does that make keepalive more clear to you ?

nickchomey commented:
Thanks, those comments are helpful re: `keepalive`. 

Though, perhaps before merging this, let me dig into the sdk today and report back later with any further feedback

nickchomey commented:
Ive done a bit of testing and it seems fine. I'll be sure to report any issues or improvements if I find them

bencroker commented:
@Superpat please resolve the conflicts and report the current status.

Superpat commented:
@bencroker done

bencroker commented:
Merged, thank you both!
